std/algorithm/comparison.d

```diff
@@ -900,6 +897,8 @@ template equal(alias pred = "a == b")
     evaluates to `bool`.
     Performs $(BIGOH min(r1.length, r2.length)) evaluations of `pred`.
 
+    At least one of the ranges must be finite. If one range involved is infinite, the result is `false`.
```

Comment:
I know that conceptually, ∞ != ∞, but it doesn't make much sense that `repeat(3) != repeat(3)`.

Reply:
Two infinite ranges won't be accepted at all. The result `false` is returned if exactly one is infinite.

Reply:
My mistake, I misread the template constraints and thought it allowed two infinite ranges.